src/Compilers/CSharp/Test/Emit/Emit/IsConstModifierTests.cs

```diff
@@ -0,0 +1,757 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Emit
+{
+    public class IsConstModifierTests : CSharpTestBase
```

Comment:
>IsConstModifierTests [](start = 17, length = 20)

We should have tests when distinct types are used as IsConst modifiers in compilation and its references. The compilation should be able to use the APIs from references, including implementing/overriding.